error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:6:22
   |
LL | fn option(i: i32) -> impl Sized { //~ ERROR
   |                      ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `std::option::Option<(impl Sized, i32)>`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:14:15
   |
LL | fn tuple() -> impl Sized { //~ ERROR
   |               ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `(impl Sized,)`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:18:15
   |
LL | fn array() -> impl Sized { //~ ERROR
   |               ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[impl Sized; 1]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:22:13
   |
LL | fn ptr() -> impl Sized { //~ ERROR
   |             ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `*const impl Sized`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:26:16
   |
LL | fn fn_ptr() -> impl Sized { //~ ERROR
   |                ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `fn() -> impl Sized`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:30:25
   |
LL | fn closure_capture() -> impl Sized { //~ ERROR
   |                         ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[closure@$DIR/recursive-impl-trait-type.rs:32:5: 32:19 x:impl Sized]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:35:29
   |
LL | fn closure_ref_capture() -> impl Sized { //~ ERROR
   |                             ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[closure@$DIR/recursive-impl-trait-type.rs:37:5: 37:20 x:impl Sized]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:40:21
   |
LL | fn closure_sig() -> impl Sized { //~ ERROR
   |                     ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[closure@$DIR/recursive-impl-trait-type.rs:41:5: 41:21]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:44:23
   |
LL | fn generator_sig() -> impl Sized { //~ ERROR
   |                       ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[closure@$DIR/recursive-impl-trait-type.rs:45:5: 45:23]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:48:27
   |
LL | fn generator_capture() -> impl Sized { //~ ERROR
   |                           ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[generator@$DIR/recursive-impl-trait-type.rs:50:5: 50:26 x:impl Sized {()}]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:53:26
   |
LL | fn substs_change<T>() -> impl Sized { //~ ERROR
   |                          ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `(impl Sized,)`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:57:24
   |
LL | fn generator_hold() -> impl Sized { //~ ERROR
   |                        ^^^^^^^^^^ resolves to self-referential type
   |
   = note: resolved type is `[generator@$DIR/recursive-impl-trait-type.rs:58:5: 62:6 {impl Sized, ()}]`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:65:40
   |
LL | async fn recursive_async_function() -> () { //~ ERROR
   |                                        ^^ resolves to self-referential type
   |
   = note: resolved type is `std::future::GenFuture<[static generator@$DIR/recursive-impl-trait-type.rs:65:43: 67:2 {impl std::future::Future, ()}]>`

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:73:26
   |
LL | fn mutual_recursion() -> impl Sync { //~ ERROR
   |                          ^^^^^^^^^ resolves to self-referential type
   |
   = note: type resolves to itself

error: recursive opaque type
  --> $DIR/recursive-impl-trait-type.rs:77:28
   |
LL | fn mutual_recursion_b() -> impl Sized { //~ ERROR
   |                            ^^^^^^^^^^ resolves to self-referential type
   |
   = note: type resolves to itself

error: aborting due to 15 previous errors

